tests/test_record.py

```diff
@@ -575,3 +570,11 @@ class MyRecordBad:
             'record_class is expected to be a subclass of asyncpg.Record',
         ):
             await self.connect(record_class=MyRecordBad)
+
+    def test_record_pickle(self):
```

Comment:
write a test of pickling nested records

Reply:
Postgres doesn't support returning nested records. asyncpg doesn't support nested records anywhere. I don't think it's a good idea.

Reply:
Nested composite types are returned as nested `Record` instances, e.g:

```python
import asyncio
import asyncpg

async def main():
    conn = await asyncpg.connect()
    await conn.execute('CREATE TYPE complex AS (r float, imag float)')
    print(await conn.fetchrow("SELECT 1, '2', (3, 4)::complex"))

asyncio.run(main())
```